src/internal/_curry1.js

```diff
@@ -10,7 +10,7 @@ module.exports = function _curry1(fn) {
   return function f1(a) {
     if (arguments.length === 0) {
       return f1;
-    } else if (a != null && a['@@functional/placeholder'] === true) {
+    } else if (a instanceof Object && a['@@functional/placeholder'] === true) {
```

Comment:
Many of the concerns raised in plaid/sanctuary#100 do not apply here, but it *is* possible to concoct a scenario we wouldn't handle correctly:

```javascript
> require('vm').runInNewContext('({"@@functional/placeholder": true})') instanceof Object
false
```